cbopensource/connectors/taxii/cybox_parse.py

```diff
@@ -71,14 +91,24 @@ def sanitize_id(id):
     return id.replace(':', '-')
 
 
-def validate_ip_address(ip_address):
+def validate_ip_address(ip_string):
     try:
-        socket.inet_aton(ip_address)
+        ipaddress.ip_address(unicode(ip_string))
         return True
-    except socket.error:
+    except ValueError:
         return False
 
 def cybox_parse_observable(observable, indicator, timestamp, score):
+    if observable.observable_composition:
+        reports = []
+        observables = list(observable.observable_composition.observables)
+        for composed_observable in observables:
```

Comment:
If `observable.observable_composition.observables` can be passed to the list constructor, then it is already an iterable type and can be used in the for loop.